File: Daily challenge/Number of Islands.py
class Solution:
    def outside(self, grid, i, j):
        if i < 0 or i >= len(grid) or j < 0 or j >= len(grid[0]):
            return True
        return False

    def DFS(self, grid, i, j):
        if not self.outside(grid, i, j) and grid[i][j] == '1':
            grid[i][j] = 'v'
        else:
            return

        self.DFS(grid, i - 1, j)
        self.DFS(grid, i + 1, j)
        self.DFS(grid, i, j - 1)
        self.DFS(grid, i, j + 1)

    def numIslands(self, grid: List[List[str]]) -> int:
        lands = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == '1':
                    lands += 1
                    self.DFS(grid, i, j)
                    
        return lands

File: Daily challenge/Number of Islands.cpp
class Solution 
{
public:
    bool outside(vector<vector<char>>& grid, int i, int j)
    {
        if(i < 0 || i >= grid.size() || j < 0 || j >= grid[0].size())
        {
            return true;
        }
        return false;
    }

    void DFS(vector<vector<char>>& grid, int i, int j)
    {
        if(!outside(grid, i, j) && grid[i][j] == '1')
        {
            grid[i][j] = 'v';
        }
        else return;

        DFS(grid, i-1, j);
        DFS(grid, i+1, j);
        DFS(grid, i, j-1);
        DFS(grid, i, j+1);
    }

    int numIslands(vector<vector<char>>& grid) 
    {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);

        int lands = 0;
        for(int i = 0; i < grid.size(); i++)
        {
            for(int j = 0; j < grid[0].size(); j++)
            {
                if(grid[i][j] == '1')
                {
                    lands++;
                    DFS(grid, i, j);        
                }
            }
        }
        return lands;
    }
};